server/startup/migrations/v185.js

```diff
@@ -0,0 +1,65 @@
+import { Mongo } from 'meteor/mongo';
+
+import { Migrations } from '../../../app/migrations/server';
+import { Settings } from '../../../app/models/server';
+import { Notification } from '../../../app/notification-queue/server/NotificationQueue';
+
+function convertNotification(notification) {
+	try {
+		const { userId } = JSON.parse(notification.query);
+		const username = notification.payload.sender?.username;
+		const roomName = notification.title !== username ? notification.title : '';
+
+		const message = roomName === '' ? notification.text : notification.text.replace(`${ username }: `, '');
+
+		return {
+			_id: notification._id,
+			uid: userId,
+			rid: notification.payload.rid,
+			ts: notification.createdAt,
+			items: [{
+				type: 'push',
+				data: {
+					roomId: notification.payload.rid,
+					payload: notification.payload,
+					roomName,
+					username,
+					message,
+					badge: notification.badge,
+					userId,
+					category: notification.apn?.category,
+				},
+			}],
+		};
+	} catch (e) {
+		//
+	}
+}
+
+Migrations.add({
+	version: 185,
+	async up() {
```

Comment:
if you want to hold until this migration is complete, you'll need to use `Promise.await()` , using `async` here will return a promise right away and the migration will process continue.

you can look how I did on migration 179:

https://github.com/RocketChat/Rocket.Chat/blob/develop/server/startup/migrations/v179.js